cluster-autoscaler/cloudprovider/hetzner/hetzner_manager.go

```diff
@@ -71,19 +94,41 @@ func newManager() (*hetznerManager, error) {
 	)
 
 	ctx := context.Background()
+	var err error
 
+	clusterConfigBase64 := os.Getenv("HCLOUD_CLUSTER_CONFIG")
 	cloudInitBase64 := os.Getenv("HCLOUD_CLOUD_INIT")
-	if cloudInitBase64 == "" {
-		return nil, errors.New("`HCLOUD_CLOUD_INIT` is not specified")
+
+	if clusterConfigBase64 == "" && cloudInitBase64 == "" {
+		return nil, errors.New("`HCLOUD_CLUSTER_CONFIG` or `HCLOUD_CLOUD_INIT` is not specified")
 	}
-	cloudInit, err := base64.StdEncoding.DecodeString(cloudInitBase64)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse cloud init error: %s", err)
+	var clusterConfig *ClusterConfig = &ClusterConfig{}
+
+	if clusterConfigBase64 != "" {
+		clusterConfig.IsUsingNewFormat = true
+	}
+
+	if clusterConfig.IsUsingNewFormat {
+		clusterConfigEnv, err := base64.StdEncoding.DecodeString(clusterConfigBase64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse cluster config error: %s", err)
+		}
+		json.Unmarshal([]byte(string(clusterConfigEnv)), &clusterConfig)
```

Comment:
`clusterConfigEnv` already is a `[]byte`, so no need to convert to string and back to `[]byte`. https://pkg.go.dev/encoding/base64#Encoding.DecodeString

```suggestion
		json.Unmarshal(clusterConfigEnv, &clusterConfig)
```

---

Also please add error handling for failed `json.Unmarshal()`.